muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/rest/TranscriptofRecordsRESTService.java

```diff
@@ -646,13 +658,30 @@ public Response listWorkspaces(
     return Response.ok(workspaces).build();
   }
   
+  private Mandatority getMandatority(SchoolDataIdentifier educationSubtypeId) {
```

Comment:
This seems to be the same method in CoursePickerRESTService, TranscriptofRecordsRESTService and WorkspaceRESTService, it should be moved to a place where it can be accessed by all so there's only one instance of the code.

In addition - for every getMandatority -call the educationTypeMapping is loaded from database. This is wasteful if it's done in a loop like it is on line 643. (Same for CoursePickerRestService line 392.)

My suggestion would be: make a getEducationMapping method in a controller (does WorkspaceEntityController make sense? not sure), which reads EducationTypeMapping from database as below and returns it. Then use EducationTypeMapping.getMandatority to ask for the state in the endpoints. Replace all three instances with this approach and the reading of the mapping can be done outside the loop where a loop exists. Also maybe check if there's more places where this generalization can be done.